src/Microsoft.AspNetCore.SystemWebAdapters/Caching/CacheDependency.cs

```diff
@@ -68,13 +68,17 @@ internal CacheDependency()
 
     protected internal void FinishInit()
     {
-        HasChanged = changeMonitors.Any(cm => cm.HasChanged && (cm.GetLastModifiedUtc() > utcStart));
-        utcLastModified = changeMonitors.Max(cm => cm.GetLastModifiedUtc());
-        if (HasChanged)
+        if (changeMonitors.Count > 0)
```

Comment:
This check is because changeMonitors count can be 0 (See `Init()`)

Reply:
wouldn't this not reset HasChanged if it has potentially been set to something? Is this a correctness issue?

Reply:
changeMonitors.Any works and sets HasChanged if changeMonitors count is 0. However, utcLastModified might have an issue if the source list is empty. However, I think that the specific overload of Max for DateTime uses the generic version which uses default(T) in the case of the empty set rather than throwing an error. In this case, it should probably use the test, and set DateTime.MinValue otherwise. I think all the other paths just short circuit (e.g. the Foreach) when there are no items so I don't believe it causes any issue.

Reply:
I've included this fix in #351